hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -495,6 +511,46 @@ spec = around (showLogsOnFailure "EndToEndSpec") $ do
             logfile <- readFileBS logFilePath
             BS.length logfile `shouldSatisfy` (> 0)
 
+    describe "forking eras" $ do
+      it "does report on unsupported era" $ \tracer -> do
+        withClusterTempDir "unsupported-era" $ \tmpDir -> do
+          args <- setupCardanoDevnet tmpDir
+          forkIntoConwayInEpoch tmpDir args 1
+          withCardanoNode (contramap FromCardanoNode tracer) defaultNetworkId tmpDir args $ \node@RunningNode{nodeSocket} -> do
+            hydraScriptsTxId <- publishHydraScriptsAs node Faucet
+            chainConfig <- chainConfigFor Alice tmpDir nodeSocket [] cperiod
+            withHydraNode' chainConfig tmpDir 1 aliceSk [] [1] hydraScriptsTxId Nothing $ \out err ph -> do
+              -- Assert nominal startup
+              waitForLog 5 out "missing NodeOptions" (Text.isInfixOf "NodeOptions")
+
+              delayEpoch tmpDir args 1
+
+              waitForProcess ph `shouldReturn` ExitFailure 1
+              hGetContents err >>= (`shouldContain` "Received blocks in not supported era.")
```

Comment:
Could: reduce the string matching and only assert what we care about. IMO that is `"unsupported era"` and `"upgrade hydra-node"` (maybe via to `shouldContain` checks).

By not matching the exact string, we break this test less likely if we rephrase or extend the message